### PR TITLE
fix(ci): update codecov token

### PR DESCRIPTION
**Pull request by JeyJeyGao**
The same issue as: https://github.com/notaryproject/notation/issues/921
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -34,4 +34,6 @@ jobs:
       - name: Run unit tests
         run: make test
       - name: Upload coverage to codecov.io
-        uses: codecov/codecov-action@v4
\ No newline at end of file
+        uses: codecov/codecov-action@v4
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (Two-Hearts) [inline .github/workflows/reusable-build.yml:39]: what's the purpose of this PR? Is there a related issue regarding this?

Comment (JeyJeyGao) [reply .github/workflows/reusable-build.yml:39]: Added issue link in the PR discription.

Comment (Two-Hearts) [reply .github/workflows/reusable-build.yml:39]: hmm, according to the doc: https://github.com/codecov/codecov-action?tab=readme-ov-file#breaking-changes:
there's a breaking change on the action, but they also stated that `PRs made from forks to the upstream public repos will support tokenless`. That means we don't need the token?

**Review by priteshbandi — APPROVED** [review]
LGTM

File: .github/workflows/reusable-build.yml (hunk 1 above)
Comment (shizhMSFT) [reply .github/workflows/reusable-build.yml:39]: Tokenless is only supported in PRs not on the `main` branch

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Pull request merged**